manifests/init.pp

```diff
@@ -7,12 +7,63 @@
 # [*ensure*]
 #   The ensure value for the package.
 #
-# [*consular_args*]
-#   A list of arguments to pass to Consular at runtime.
+# [*host*]
+#   The address for Consular to listen on for connections from Marathon.
+#
+# [*port*]
+#   The port for Consular to listen on for connections from Marathon.
+#
+# [*consul*]
+#   The full address to use to connect to Consul.
+#
+# [*marathon*]
+#   The full address to use to connect to Marathon.
+#
+# [*registration_id*]
+#   The registration ID to use when registering for callbacks with Marathon.
+#
+# [*sync_interval*]
+#   The number of seconds between syncs. If 0, syncing will be disabled.
+#
+# [*purge*]
+#   Whether or not to purge old Consular-registered services during syncs.
+#
+# [*opts*]
+#   A hash of additional options to pass to Consular. NOTE: these options can
+#   override any of the above options.
+#   e.g. { 'timeout' => 10 }
```

Comment:
Maybe call this `extra_opts` or `extra_args` instead? `opts` sounds like another parameter rather than a catchall for anything we don't have a specific parameter for.